solr/core/src/java/org/apache/solr/core/SolrResourceLoader.java

```diff
@@ -954,4 +987,46 @@ public static void persistConfLocally(SolrResourceLoader loader, String resource
     }
   }
 
+  // TODO document these methods...
```

Comment:
What is the motivation behind `SolrResourceLoader` returning packages?

Reply:
I came to think of the SolrResourceLoader as being rather related to packages.  SRL has methods that lookup class names to actual Classes and then instances, so the focal point of my approach is to have `findClass` here work with the package system.

Reply:
SRL has nothing to do with packages. It's an anti pattern. SRL is used by PackageLaoder and SRL is totally agnostic of the package system. We should just limit exposure of Packages to 2 places
- `CoreContainer` : This is where the package loader is stored
- `SolrCore` : This is where the correct version of a package for a plugin can be identified and listeners for package updates are registered

Without the correct version, we leave the system inconsistent

Reply:
The SRL is pervasively used across the codebase to load stuff (hence "resource loader" in its name).  If CoreContainer/SolrCore is now the right place, then this is a big change; no?

BTW, Wouldn't the ConfigSet object be a more befitting place to store the version association than the SolrCore?  Isn't params.json (which is where the association is) stored in the config set?